pandas/core/arrays/datetimelike.py

```diff
@@ -347,21 +326,59 @@ def _validate_frequency(cls, index, freq, **kwargs):
     # Arithmetic Methods
 
     def _add_datelike(self, other):
+        # Overriden by TimedeltaArray
         raise TypeError("cannot add {cls} and {typ}"
                         .format(cls=type(self).__name__,
                                 typ=type(other).__name__))
 
+    _add_datelike_dti = _add_datelike
+
     def _sub_datelike(self, other):
```

Comment:
after reading your other comments, I guess the _dti are simply the array-like versions? if so, then these really really need to be more explicit, e.g. something like

```
_add_datetimelike_array
_add_datelike_scalar
```

if not, then the others needs to be very very clear via comments & types

Reply:
> I guess the _dti are simply the array-like versions?

Yes.  There is an existing `_add_delta_tdi`, so this followed that pattern.  Making the names more explicit sounds good to me.